backend/api/pipeline.proto

```diff
@@ -136,6 +137,29 @@ message GetTemplateResponse {
   string template = 1;
 }
 
+message GetPipelineVersionTemplateRequest{
+  string pipeline_id = 1;
+  string version_id = 2;
+}
+
+message CreatePipelineVersionRequest {
+  string pipeline_id = 1;
+  PipelineVersion version = 2;
+}
+
+message GetPipelineVersionRequest {
+  string pipeline_id = 1;
+  string version_id = 2;
+}
+
+message ListPipelineVersionsRequest {
```

Comment:
we should consider adding pagination for list version.

Reply:
Done.